File: app/routers/exchange_rate.py
from fastapi import APIRouter
from fastapi.responses import JSONResponse
from ..services.currency_service import CurrencyService

router = APIRouter()

@router.get("/exchange-rate")
async def exchange_rate():
    try:
        service = CurrencyService()
        rate = service.get_dollar_to_real_rate()
        return JSONResponse(content={
            "success": True,
            "rate": rate,
            "formatted_rate": f"{rate:,.2f}".replace(",", "X").replace(".", ",").replace("X", "."),
            "currency": "BRL"
        })
    except Exception:
        return JSONResponse(status_code=500, content={
            "success": False,
            "error": "Erro ao obter taxa de câmbio",
            "rate": 5.00
        })

File: backup_php/public/api/exchange-rate.php
<?php
ini_set('display_errors', 0);
error_reporting(0);

// Carrega o autoloader do Composer
require_once __DIR__ . '/../../vendor/autoload.php';

// Carrega as variáveis de ambiente
$envPath = __DIR__ . '/../../.env';
if (file_exists($envPath)) {
    $dotenv = Dotenv\Dotenv::createImmutable(__DIR__ . '/../../');
    $dotenv->load();
}

use App\Services\CurrencyService;

header('Content-Type: application/json');
header('Access-Control-Allow-Origin: *');
header('Access-Control-Allow-Methods: GET, OPTIONS');
header('Access-Control-Allow-Headers: Content-Type');

if ($_SERVER['REQUEST_METHOD'] === 'OPTIONS') {
    exit(0);
}

try {
    $currencyService = new CurrencyService();
    $rate = $currencyService->getDollarToRealRate();
    
    echo json_encode([
        'success' => true,
        'rate' => $rate,
        'formatted_rate' => number_format($rate, 2, ',', '.'),
        'currency' => 'BRL'
    ]);
    
} catch (Exception $e) {
    http_response_code(500);
    echo json_encode([
        'success' => false,
        'error' => 'Erro ao obter taxa de câmbio',
        'rate' => 5.00 // Fallback
    ]);
}
